R/simulation.R

```diff
@@ -82,7 +82,7 @@ Simulation <- R6::R6Class(
     molWeightFor = function(quantityPath) {
       ospsuite.utils::validateIsString(quantityPath)
       mw <- rClr::clrCall(self$ref, "MolWeightFor", quantityPath)
-      mw %||% NA
+      mw %||% NA_real_

```

Comment:
What is this ? Real NA?

Reply:
`NA` of real type, yes. The default NA in R is logical, so when we know the expected data type, we can use the specific variant of it. R usually does this coercion implicitly, so you never need to worry about it.

``` r
is.logical(NA)
#> [1] TRUE
is.character(NA_character_)
#> [1] TRUE
is.double(NA_real_)
#> [1] TRUE
is.complex(NA_complex_)
#> [1] TRUE
is.integer(NA_integer_)
#> [1] TRUE
```

<sup>Created on 2022-01-21 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

Reply:
Oh my...I have used NA in a few places in the code ....maybe we should do a quick search for other instances?

Just for me:
Why do we care to have NA typed? Isn't NA a short cut for not available?

Reply:
> Why do we care to have NA typed? Isn't NA a short cut for not available?

We actually needn't care. It's just that my OCD acts up every time I see that the output type is inconsistent with what the users expect.

For example, when I convert simResults object to a dataframe, the `molWeight` column for molecular weight can be `NA` if the weights can't be extracted from the object. But the `NA` is of `logical` type (`<lgl>`), when, as a user, I would expect a `double`.

``` r
library(ospsuite)
#> Loading required package: rClr
#> Loading the dynamic library for Microsoft .NET runtime...
#> Loaded Common Language Runtime version 4.0.30319.42000

simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
sim <- loadSimulation(simPath)
results <- runSimulations(sim)
df <- simulationResultsToDataFrame(results)

head(dplyr::select(df, molWeight))
#> # A tibble: 6 x 1
#>   molWeight
#>   <lgl>    
#> 1 NA       
#> 2 NA       
#> 3 NA       
#> 4 NA       
#> 5 NA       
#> 6 NA
```

<sup>Created on 2022-01-21 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

Reply:
Ok I get it now.. I hate R always trying to cast data types without any strict rules..

Reply:
Dynamic type checking 🤷‍♂️